include/pms-utils/misc/readfile.hpp

```diff
@@ -0,0 +1,22 @@
+#include <cerrno>
+#include <expected>
+#include <filesystem>
+#include <fstream>
+#include <string>
+
+namespace pms_utils::misc {
+
+std::expected<std::string, int> try_readfile(std::filesystem::path &path) {
```

Comment:
I don't see a good reason to return `std::expected` here. The failure modes of `ifstream` and `istreambuf_iterator` are 100% impl-defined, with no standardized exceptions or errno values (or even any guarantee of errno being set). We also don't really care about recovering from OS errors in a meaningful way. Perhaps just return `std::optional` ?